### DISCUSSION TITLE
installing a module(library) to a specific directory

### DISCUSSION THREAD

**Discussion by binup28** [pybind/pybind11#3278] [Q&A]
I am trying to have my pybind11 module(MyModule) installed into my BUILD_DIR directory. I tried "install", "set_target_properties" - but neither works and it didnt even throw errors. It always installs the so file in the current directory rather than BUILD_DIR.

pybind11_add_module(MyModule MyFile.cpp)
install( TARGETS MyModule DESTINATION "${BUILD_DIR}" )
set_target_properties( MyModule PROPERTIES RUNTIME_OUTPUT_DIRECTORY "${BUILD_DIR}" )

How can I fix this?

**Comment by hbadi** [comment]
I'm new with pybind11 and I also have the problem...
Moreover i can't understand, even when i lauch the interpreter from the build directory, how python can import a module just using the so file, it's a mystery.

**Comment by LivelyLiz** [comment]
Bit late, but have you tried setting CMAKE_LIBRARY_OUTPUT_DIRECTORY? This works for me. Also even before I used that, it was placed it in the build directory with setting CMAKE_BINARY_DIR. I think both could work for a single target if you set LIBRARY_OUTPUT_DIRECTORY or BINARY_DIR for just that target.
